code/modules/surgery/repair_puncture.dm

```diff
@@ -0,0 +1,111 @@
+
+/////BURN FIXING SURGERIES//////
+
+#define REALIGN_INNARDS 1
+#define WELD_VEINS		2
+
+///// Repair puncture wounds
+/datum/surgery/repair_puncture
+	name = "Repair puncture"
+	steps = list(/datum/surgery_step/incise, /datum/surgery_step/repair_innards, /datum/surgery_step/seal_veins, /datum/surgery_step/close) // repeat between steps 2 and 3 until healed
+	target_mobtypes = list(/mob/living/carbon)
+	possible_locs = list(BODY_ZONE_R_ARM,BODY_ZONE_L_ARM,BODY_ZONE_R_LEG,BODY_ZONE_L_LEG,BODY_ZONE_CHEST,BODY_ZONE_HEAD)
+	requires_real_bodypart = TRUE
+	targetable_wound = /datum/wound/pierce
+	var/next_step = REALIGN_INNARDS
+
+/datum/surgery/repair_puncture/can_start(mob/living/user, mob/living/carbon/target)
+	if(..())
+		var/obj/item/bodypart/targeted_bodypart = target.get_bodypart(user.zone_selected)
+		var/datum/wound/burn/pierce_wound = targeted_bodypart.get_wound_type(targetable_wound)
+		return(pierce_wound && pierce_wound.blood_flow > 0)
+
+//SURGERY STEPS
+
+///// realign the blood vessels so we can reweld them
+/datum/surgery_step/repair_innards
+	name = "realign blood vessels"
+	implements = list(TOOL_HEMOSTAT = 100, TOOL_SCALPEL = 85, TOOL_WIRECUTTER = 40)
```

Comment:
it'd be nice to have ghetto options for this that AREN'T engineering tools, for prisoners and such. maybe a fork?